app/controllers/purchases_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PurchasesController < ApplicationController
+  before_action :logged_in_user, only: %i(create new)
+  before_action :cart_is_empty?, only: :create
+
+  def new
+    @purchase = Purchase.new
+  end
+
+  def create
+    @purchase = current_user.purchases.build(purchase_params)
+    if @purchase.save
+      add_products_to_detailspurchase @purchase.id
```

Comment:
em tìm hiểu về `transaction in rails ` nhé, vì hiện tại sẽ có nhìu câu lệnh làm change DB nè

Reply:
Dạ ok chị cám ơn chị ạ